src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsMapping.java

```diff
@@ -0,0 +1,255 @@
+        boolean isYes = YesNo.isYes(hearingsInWelshFlag);
+        return isYes ? YES : NO;
+    }
+
+    public static YesNo shouldBeAdditionalSecurityFlag(YesNo additionalSecurityFlag) {
+        boolean isYes = YesNo.isYes(additionalSecurityFlag);
+        // TODO Check unacceptableCustomerBehaviour for Appellant, their Appointee and their Representatives
+        //      Check unacceptableCustomerBehaviour for each OtherParty, their Appointee and their Representatives
+        //      Any YES then YES
+        return isYes ? YES : NO;
+    }
+
+    public static YesNo shouldBeSensitiveFlag(YesNo sensitiveFlag) {
+        boolean isYes = YesNo.isYes(sensitiveFlag);
+        // TODO check no other factors
+        return isYes ? YES : NO;
+    }
+
+    public static String getCaseDeepLink(String ccdCaseId) {
+        // TODO Confirm this is correct and create automated tests
+        return String.format("%s/cases/case-details/%s", exUiUrl, ccdCaseId);
+    }
+
+    public static String getCaseManagementLocationCode(CaseManagementLocation caseManagementLocation) {
+        // TODO find out how to get epims for this using caseManagementLocation
+        return null;
+    }
+
+    public static List<CcdValue<String>> getFacilitiesRequired(SscsCaseData caseData) {
+        List<CcdValue<String>> facilitiesRequired = new ArrayList<>();
+        // TODO find out how to work this out and implement
+        return facilitiesRequired;
+    }
+
+    public static HearingWindowRange getHearingWindowRange(SscsCaseData caseData) {
+        // TODO check this is correct and if any additional logic is needed
+        if (YesNo.isYes(caseData.getAutoListFlag()) && nonNull(caseData.getEvents())) {
+            Event dwpResponded = caseData.getEvents().stream()
+                    .filter(c -> EventType.DWP_RESPOND.equals(c.getValue().getEventType()))
+                    .findFirst().orElse(null);
+            if (nonNull(dwpResponded) && nonNull(dwpResponded.getValue())) {
+                LocalDate hearingWindowStart = dwpResponded.getValue().getDateTime().plusMonths(1).toLocalDate();
+                return HearingWindowRange.builder().dateRangeStart(hearingWindowStart).build();
+            }
+        }
+        return null;
+    }
+
+    public static LocalDateTime getFirstDateTimeMustBe(SscsCaseData caseData) {
+        LocalDateTime firstDateTimeMustBe = null;
+        // TODO Find out how to use adjournCase data to work this out, possibly related variables:
+        //      adjournCaseNextHearingDateType, adjournCaseNextHearingDateOrPeriod, adjournCaseNextHearingDateOrTime,
+        //      adjournCaseNextHearingFirstAvailableDateAfterDate, adjournCaseNextHearingFirstAvailableDateAfterPeriod
+        return firstDateTimeMustBe;
+    }
+
+    public static PanelRequirements getPanelRequirements(SscsCaseData caseData) {
+        PanelRequirements panelRequirements = PanelRequirements.builder().build();
```

Comment:
Can use builder methods when setting each of the properties then use `.build()` at the return step